### PR TITLE
feat: enable snyk security scanning

### PR DESCRIPTION
**Pull request by p0tr3c**
Attempt to add snyk scanning for PRs that have been submitted by maintainers.

The contributor PRs (from forked repos) should be ignored by the filter.

### CODE REVIEW

**Comment by p0tr3c** [comment]
@eliecharra  you have probably looked at this problem. Does this address the issue of running snyk CLI on contributor PRs sufficiently?

**Comment by p0tr3c** [comment]
I haven't worked on open source repos and circleci. But it appears that circle does not run on the PRs from my forked repo at all. Is that the case?

**Comment by p0tr3c** [comment]
I've tested this on the forked PR and it appears that filter is skipping the scan for contributors. We would need to catch any issues in merge to main. 
This has some consequences, mainly if the PR includes new vulns we will have to issue fix PRs. The alternative is to report but not fail the build. Opinions of what feels more preferable?

**Comment by p0tr3c** [comment]
You can see the example outputs for contributor PR here https://github.com/snyk/lambda-env-updater/pull/3. Go to circle CI to see various run results for this job.

**Comment by p0tr3c** [comment]
Nice @eliecharra  this is exactly the feedback I was looking for. This is for sure a workaround for our internal security needs.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,4 +1,17 @@
 version: 2.1
+only_maintainers: &only_maintainers
+  filters:
+    branches:
+      # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
+      ignore: /pull\/[0-9]+/
```
Comment (eliecharra) [inline .circleci/config.yml:6]: I'm fine with that (it came from the official circleCI doc) 👍🏻

**Comment by eliecharra** [comment]
Would you mind to push that same chunk of diff to driftctl too @p0tr3c 🙏🏻

**Comment by p0tr3c** [comment]
> Would you mind to push that same chunk of diff to driftctl too @p0tr3c 🙏🏻

I will. Thanks

**Pull request merged**